pkg/apis/batch/validation/validation.go

```diff
@@ -181,7 +181,6 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 	if spec.TTLSecondsAfterFinished != nil {
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*spec.TTLSecondsAfterFinished), fldPath.Child("ttlSecondsAfterFinished"))...)
 	}
-	// CompletionMode might be nil when IndexedJob feature gate is disabled.
 	if spec.CompletionMode != nil {
```

Comment:
Although `spec.CompletionMode` is expected to be initialized by `SetDefaults_Job()`:
https://github.com/kubernetes/kubernetes/blob/6820a383be4a1bbb5cd526eb3aeb9e6ade351498/pkg/apis/batch/v1/defaults.go#L47-L50

It is always a good practice to check whether a pointer is nil, just like we did with the other fields.

For example: 
https://github.com/kubernetes/kubernetes/blob/6820a383be4a1bbb5cd526eb3aeb9e6ade351498/pkg/apis/batch/v1/defaults.go#L40-L42
https://github.com/kubernetes/kubernetes/blob/6820a383be4a1bbb5cd526eb3aeb9e6ade351498/pkg/apis/batch/validation/validation.go#L178-L180

Reply:
The reason is that an old Job that was never updated might have a nil CompletionMode.

But I don't think we need a comment for that :)